frontend/src/components/chat/ChatInterface.test.tsx

```diff
@@ -9,17 +9,58 @@ import ActionType from "#/types/ActionType";
 import { addAssistantMessage } from "#/state/chatSlice";
 import AgentState from "#/types/AgentState";
 
+/// <reference types="vitest" />
+
+interface CustomMatchers<R = unknown> {
+  toMatchMessageEvent(expected: string): R;
+}
+
+declare module "vitest" {
+  interface Assertion<T> extends CustomMatchers<T> {}
+  interface AsymmetricMatchersContaining extends CustomMatchers {}
+}
+
 // This is for the scrollview ref in Chat.tsx
 // TODO: Move this into test setup
 HTMLElement.prototype.scrollTo = vi.fn().mockImplementation(() => {});
+const TEST_TIMESTAMP = new Date().toISOString();
 
 describe("ChatInterface", () => {
   const sessionSendSpy = vi.spyOn(Session, "send");
   vi.spyOn(Session, "isConnected").mockReturnValue(true);
 
+  expect.extend({
+    toMatchMessageEvent(received, expected) {
+      const receivedObj = JSON.parse(received);
+      const expectedObj = JSON.parse(expected);
+
+      // Compare everything except the timestamp
+      const { timestamp: receivedTimestamp, ...receivedRest } =
+        receivedObj.args;
+      const { timestamp: expectedTimestamp, ...expectedRest } =
+        expectedObj.args;
+
+      const pass =
+        this.equals(receivedRest, expectedRest) &&
+        typeof receivedTimestamp === "string";
+
+      return {
+        pass,
+        message: () =>
+          pass
+            ? `expected ${received} not to match the structure of ${expected} (ignoring exact timestamp)`
+            : `expected ${received} to match the structure of ${expected} (ignoring exact timestamp)`,
+      };
+    },
+  });
+
```

Comment:
Is this for testing timestamps?

Reply:
> Is this for testing timestamps?

Yes, I'm not sure how to best go about it, if you have any recommendations?

Reply:
Maybe mocking timers could help instead:
- https://vitest.dev/guide/mocking#timers
- https://vitest.dev/api/vi.html#vi-usefaketimers

Reply:
I'd prefer to not spend more hours on just this as this is a working solution, tbh.

Reply:
Sure, will do that next.